frontend/src/modules/widget/widget-utility.js

```diff
@@ -0,0 +1,93 @@
+import {
+  ALL_TIME_PERIOD_FILTER,
+  WIDGET_PERIOD_OPTIONS,
+  WIDGET_GRANULARITY_OPTIONS,
+  WEEKLY_GRANULARITY_FILTER,
+  SEVEN_DAYS_PERIOD_FILTER,
+  MONTHLY_GRANULARITY_FILTER,
+  FOURTEEN_DAYS_PERIOD_FILTER,
+  THREE_MONTHS_PERIOD_FILTER,
+} from './widget-constants';
+
+export const getSelectedPeriodFromLabel = (
+  label,
+  defaultMinimumPeriod,
+  granularity
+) => {
+  if (label === 'All time') return ALL_TIME_PERIOD_FILTER;
+
+  let selectedPeriod = WIDGET_PERIOD_OPTIONS.find(
+    (option) => option.label === label
+  );
+
+  if (granularity) {
+    selectedPeriod = getGranularityBasedPeriod(granularity, selectedPeriod);
+  }
+
+  if (selectedPeriod && comparePeriods(selectedPeriod, defaultMinimumPeriod)) {
+    return selectedPeriod;
+  }
+
+  return defaultMinimumPeriod;
+};
+
+export const getSelectedGranularityFromLabel = (
+  label,
+  defaultMinimumGranularity
+) => {
+  const selectedGranularity = WIDGET_GRANULARITY_OPTIONS.find(
+    (option) => option.label === label
+  );
+
+  if (
+    selectedGranularity &&
+    compareGranularityValues(
+      selectedGranularity.value,
+      defaultMinimumGranularity.value
+    )
+  ) {
+    return selectedGranularity;
+  }
+
+  return defaultMinimumGranularity;
+};
+
+const compareGranularityValues = (a, b) => {
```

Comment:
Can you reorder the methods so that methods are not used before they are defined?
There are also some eslint errors in this file, can you run `npm run lint` on the `frontend` folder to check which parts need fixing? 🙏 

If you check, the PRs checks failed because of lint issues

Reply:
Hey @joanagmaia thanks for the review was using prettier :), have resolved the lint errors